File: 241021/숫자 카운트/numeric-count.cpp
#include<iostream>
using namespace std;
int main()
{
    int n,ans=0;
    cin>>n;
    for(int v=0;v<n;v++)
    {
        int a,b,c;
        cin>>a>>b>>c;
        for(int i=1;i<=9;i++)
        {
            for(int j=1;j<=9;j++)
            {
                for(int k=1;k<=9;k++)
                {
                    if(i == j || i == k || j == k) continue;
                    int cnt1=0,cnt2=0,x=a/100,y=a%100/10,z=a%10;
                    if(x==i) cnt1++;
                    if(y==j) cnt1++;
                    if(z==k) cnt1++;
                    if(x==j || x==k) cnt2++;
                    if(y==i || y==k) cnt2++;
                    if(z==i || z==j) cnt2++;

                    if(cnt1 == b && cnt2 == c) ans++;
                }
            }
        }
    }
    cout<<ans;
    return 0;
}

File: 241025/숫자 카운트/numeric-count.py
n = int(input())

arr_2d = [tuple(map(int, input().split())) for _ in range(n)]

cnt = 0

for i in range(1,10):
    for j in range(1,10):
        for k in range(1,10):
            if i == j or j == k or k == i:
                continue
            one_cnt = 0
            two_cnt = 0
            success = True
            for num, cnt1,cnt2 in arr_2d:
                a = num // 100
                b = num // 10 % 10
                c = num % 10
            
                if i == a or j == b or k == c:
                    one_cnt += 1
                elif i in num or j in num or k in num:
                    two_cnt += 1
            
                if one_cnt != cnt1 and two_cnt != cnt2:
                    success =False
            if success :
                cnt += 1

print(cnt)